### PR TITLE
Add in-app storage capability

### PR DESCRIPTION
**Pull request by shayanpourvatan**

### CODE REVIEW

File: auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 
     private const val BAZAAR_WITH_AUTH_VERSION = 801500
 
     fun isBazaarInstalledOnDevice(context: Context): Boolean {
-        return getPackageInfo(context, BAZAAR_PACKAGE_NAME) != null &&
+        return getPackageInfo(
+            context,
+            BAZAAR_PACKAGE_NAME
+        ) != null &&
                 Security.verifyBazaarIsInstalled(context)
```
Comment (PHELAT) [inline auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt:18]:
```suggestion
        return getPackageInfo(context, BAZAAR_PACKAGE_NAME) != null && 
            Security.verifyBazaarIsInstalled(context)
```
This looks better IMO.

File: auth/src/main/java/com/farsitel/bazaar/storage/connection/ReceiverStorageConnection.kt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            ACTION_STORAGE_SET_DATA,
+            ACTION_STORAGE_GET_DATA
+            -> {
+                context.sendBroadcast(
+                    getNewIntentForBroadcast(
+                        requireNotNull(intent.action),
+                        intent.extras
+                    )
+                )
+            }
```
Comment (PHELAT) [inline auth/src/main/java/com/farsitel/bazaar/storage/connection/ReceiverStorageConnection.kt:45]:
```suggestion
            ACTION_STORAGE_SET_DATA,
            ACTION_STORAGE_GET_DATA -> {
                context.sendBroadcast(
                    getNewIntentForBroadcast(
                        requireNotNull(intent.action),
                        intent.extras
                    )
                )
            }
```
This looks better IMO.

Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/storage/connection/ReceiverStorageConnection.kt:45]: IDE suggestion

File: auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @JvmStatic
+    fun saveData(
+        context: Context,
+        owner: LifecycleOwner?,
+        data: ByteArray,
+        callback: BazaarStorageCallback
+    ) {
+        StorageConnection.getStorageConnection(context).savedData(owner, data, callback)
```
Comment (hamidrezasahraei) [inline auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt:38]: Is this name correct? this method name is `saveData` but it actually calls `savedData`

File: auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    fun saveDataSync(
+        context: Context,
+        owner: LifecycleOwner?,
+        data: ByteArray
+    ) {
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            throw IllegalStateException("Can't call this method on UI thread.")
+        }
+
+        StorageConnection.getStorageConnection(context).savedDataSync(owner, data)
```
Comment (hamidrezasahraei) [inline auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt:51]: Same here

File: auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt (hunk 1 above)
Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt:18]: Done

File: auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt (hunk 3 above)
Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt:38]: Done

File: auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt (hunk 4 above)
Comment (shayanpourvatan) [reply auth/src/main/java/com/farsitel/bazaar/storage/BazaarStorage.kt:51]: Done

File: auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt [hunk 5]
```diff
@@ -1,22 +1,25 @@
...
-import com.farsitel.bazaar.auth.util.getPackageInfo
-import com.farsitel.bazaar.auth.util.versionCodeSDKAware
 import com.farsitel.bazaar.auth.view.BazaarInstallerActivity
+import com.farsitel.bazaar.util.getPackageInfo
+import com.farsitel.bazaar.util.versionCodeSDKAware
 
 object BazaarClientProxy {
 
     private const val BAZAAR_WITH_AUTH_VERSION = 801500
+    private const val BAZAAR_WITH_STORAGE_VERSION = 801703
```
Comment (hamidrezasahraei) [inline auth/src/main/java/com/farsitel/bazaar/BazaarClientProxy.kt:12]: Update this

**Pull request merged**